splunk_add_on_ucc_framework/commands/openapi_generator/ucc_to_oas.py

```diff
@@ -0,0 +1,352 @@
+                    ),
+                },
+            )
+        },
+    )
+
+
+def __get_output_mode() -> Dict[str, Any]:
+    return {
+        "name": "output_mode",
+        "in": "query",
+        "required": False,
+        "description": "The name of the item to operate on",
+        "schema": {"type": "string", "enum": ["xml", "json"]},
+    }
+
+
+def __assign_ta_paths(
+    *,
+    open_api_object: OpenAPIObject,
+    path: str,
+    path_name: str,
+    actions: List[str],
+):
+    if open_api_object.paths is not None:
+        open_api_object.paths[path] = oas.PathItemObject(
+            get=__get_path_get_for_list(name=path_name),
+            post=__get_path_post_for_create(name=path_name),
+        )
+        open_api_object.paths[path].parameters = [
+            __get_output_mode(),
+        ]
+        if actions is not None and "clone" in actions:
+            path = f"{path}/{{name}}"
+            open_api_object.paths[path] = oas.PathItemObject(
+                get=__get_path_get_for_item(name=path_name),
+                post=__get_path_post_for_update(name=path_name),
+                delete=__get_path_delete(name=path_name)
+                if "delete" in actions
+                else None,
+            )
+            open_api_object.paths[path].parameters = [
+                {
+                    "name": "name",
+                    "in": "path",
+                    "required": True,
+                    "description": "The name of the item to operate on",
+                    "schema": {"type": "string"},
+                },
+                __get_output_mode(),
+            ]
+    return open_api_object
+
+
+def __add_paths(
+    open_api_object: OpenAPIObject, global_config: DataClasses
+) -> OpenAPIObject:
+    for tab in global_config.pages.configuration.tabs:
+        open_api_object = __assign_ta_paths(
+            open_api_object=open_api_object,
+            path=f"/{global_config.meta.restRoot}_settings/{tab.name}"
+            if tab.name in ["logging", "proxy"]
+            else f"/{global_config.meta.restRoot}_{tab.name}",
+            path_name=tab.name,
+            actions=tab.table.actions
+            if hasattr(tab, "table") and hasattr(tab.table, "actions")
+            else None,
+        )
+    if hasattr(global_config.pages, "inputs") and hasattr(
+        global_config.pages.inputs, "services"
+    ):
+        for service in global_config.pages.inputs.services:
+            open_api_object = __assign_ta_paths(
+                open_api_object=open_api_object,
+                path=f"/{global_config.meta.restRoot}_{service.name}",
+                path_name=service.name,
+                actions=global_config.pages.inputs.table.actions,
+            )
+    return open_api_object
+
+
+def transform(
+    global_config: global_config_lib.GlobalConfig,
+    app_manifest: app_manifest_lib.AppManifest,
+) -> OpenAPIObject:
+    content_copy = copy.deepcopy(global_config.content)
```

Comment:
I was trying with lru_cache for the content and no deepcopy here, but were causing smoke test failures like "FAILED tests/smoke/test_ucc_build.py::test_ucc_generate_with_inputs_configuration_alerts - TypeError: 'DataClasses' object is not subscriptable", so gave up, despite I'm not a big fun of deepcopy

Reply:
I was having the same error when I did not use deepcopy.